servicex_app/servicex/code_gen_adapter.py

```diff
@@ -48,16 +47,44 @@ def generate_code_for_selection(
         from io import BytesIO
         from zipfile import ZipFile
 
+        postObj = {}
+        if transformer_image:
+            postObj = {
+                "code": request_record.selection,
+                "transformer_image": transformer_image,
+            }
+        else:
+            postObj = {
+                "code": request_record.selection,
+            }
+
         result = requests.post(self.code_gen_url + "/servicex/generated-code",
-                               data=request_record.selection)
+                               json=postObj)
 
         if result.status_code != 200:
             msg = result.json()['Message']
             raise ValueError(f'Failed to generate translation code: {msg}')
 
-        zipfile = ZipFile(BytesIO(result.content))
+        print(result.data)
+        boundary = result.data[2:34].decode('utf-8')
```

Comment:
What is going on here? What is this magic _2:34_ magic number?

Reply:
I am using a python package called requests_toolbelt for managing the multipart mime encoding and decoding.
This package adds a pseudo-random sequence of characters before and after the body and also between each multipart mime field. 2:34 is where the first boundary found in all the requests, for decoding the multipart mime the decoder requires the boundary, hence the magic number is needed.

Reply:
Please investigate further. I can't believe this is the correct way to interpret a multipart-mime response